tests/sonarqube_standard_dynamic_test.go

```diff
@@ -0,0 +1,135 @@
+package tests
+
+import (
+	"crypto/tls"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+	"log"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	http_helper "github.com/gruntwork-io/terratest/modules/http-helper"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Teardown resources after test execution: namespace deleted
+func setupTest(t *testing.T) func(t *testing.T, options *k8s.KubectlOptions, namespaceName string) {
+	log.Println("setup test: No specific actions")
+
+	return func(t *testing.T, options *k8s.KubectlOptions, namespaceName string) {
+		log.Println("teardown test")
+		k8s.DeleteNamespace(t, options, namespaceName)
+	}
+}
+
+// This test checks the dynamic compatibility test for the sonarqube standard and dce charts
+// Specifically, we perform tests that:
+// - check if the required (default) number of pods is running
+// - check if all the running pods have the containersReady condition set to true
+// - check if the sonarqube application is ready (apis are ready to serve new requests)
+func TestSonarQubeChartDynamicCompatibility(t *testing.T) {
+	
+	// Input values for our charts' tests
+	table := []struct {
+		name     string
+		chartName    string
+		expectedPods	int
+		values   map[string]string
+	}{
+		{"standard-chart", "sonarqube", 2, map[string]string{"tests.enabled": "false",
+		//"image.repository": "sonarqube-arm64-v8", // DO NOT SET: This is only an experimental setting for M1 users 
+		}},
+		{"dce-chart", "sonarqube-dce", 6, map[string]string{"tests.enabled": "false",
+		//"searchNodes.image.repository": "sonarqube-arm64-v8", // DO NOT SET: This is only an experimental setting for M1 users 
+		//"ApplicationNodes.image.repository": "sonarqube-arm64-v8", // DO NOT SET: This is only an experimental setting for M1 users 
+		"ApplicationNodes.jwtSecret": "dZ0EB0KxnF++nr5+4vfTCaun/eWbv6gOoXodiAMqcFo=",
+		}}, 
+	}
+
+
+	for _, tc := range table {
+		t.Run(tc.name, func(t *testing.T) { 
+
+	teardownTest := setupTest(t)
+
+	chartName := tc.chartName
+	values := tc.values
+	expectedPods := tc.expectedPods
+
+	fmt.Printf("Value is %v", values["tests.enabled"])
+
+	// Path to the helm chart we will test
+	helmChartPath := "../charts/" + chartName
+
+	// Setup the kubectl config and context. Here we choose to use the defaults, which is:
+	// - HOME/.kube/config for the kubectl config file
+	// - Current context of the kubectl config file
+	existingKubectlOptions := k8s.NewKubectlOptions("", "", "default")
+
+	// Define a new namespace
+	namespaceName := chartName + "-dynamic-test"
+	k8s.CreateNamespace(t, existingKubectlOptions, namespaceName)
```

Comment:
This fail if the ns already exist ( which can happen in case of previous run error etc )

On the setup of the test maybe we could ensure the ns does not exist ? it would also prevent leftovers ?